pagopa/io-dev-api-server
# feat: [IOBP-687,IOBP-886] Add the recent used payment method when the user pays

## Short description
This PR updates the ecommerce and pagopa wallet APIs definitions to handle even the recent used payment method when the user pays.

## List of changes proposed in this pull request
- Added the API `/user/lastPaymentMethodUsed`;
- Added inside the WalletDB the recent used payment method that is saved in a variable that is in the RAM lifetime;
- Added a logic to save in the variable the most recently used payment method when the API `POST /transactions/:transactionId/auth-requests` is called;

## How to test
- Checkout this PR on the io-app to test the flow and check that the app selects the correct recent used payment method when you complete a payment flow inside the app;
